kedro/framework/session/session.py

```diff
@@ -209,6 +210,7 @@ def _setup_logging(self) -> None:
         """Register logging specified in logging directory."""
         conf_logging = self._get_logging_config()
         logging.config.dictConfig(conf_logging)
+        configure_logging(conf_logging)
```

Comment:
This to me feels weird. If we're getting rid of `get_current_session` altogether, it'll mean everytime someone creates a session (even just to get the store or the context - if that'll still be a thing), they'll re-configure logging. In my head this was done once, at the top of the workflow, when we `bootstrap_project()`.

Reply:
@idanov

Reply:
That's ok though, since this is the current behaviour anyways if you aren't using the `get_current_session`. We cannot do it at `bootstrap_project()` time because that's done only for non-packaged projects and at the time `bootstrap_project()` is called, we have no idea yet how and where to load the logging config from.

The logging config is setup only later at runtime on purpose, e.g. when deploying your code, theoretically you might need to change the logging config to suit your set of tools in production that consume your logs. The only lifecycle that is guaranteed to exist at runtime is the session initialisation, so it feels natural to put the config setup there.

Reply:
Ah fair enough thanks for the explanation!